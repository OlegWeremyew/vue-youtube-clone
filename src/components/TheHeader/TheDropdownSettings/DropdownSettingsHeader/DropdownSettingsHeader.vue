<template>
  <section class="flex border-b border-gray-200 text-black">
    <button
        type="button"
        class="px-3 focus:outline-none"
        @click.stop="$emit('back')"
    >
      <BaseIcon name="arrowLeft"/>
    </button>
    <span class="py-3">{{title}}</span>
  </section>
</template>

<script>
import BaseIcon from "../../../Common/BaseIcon/BaseIcon.vue";

export default {
  name: "TheDropdownSettingsHeader",
  components: {BaseIcon},
  props: ['title'],
  emits: ['back'],
}
</script>