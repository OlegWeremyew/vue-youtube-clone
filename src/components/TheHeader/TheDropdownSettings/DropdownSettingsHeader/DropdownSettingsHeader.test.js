import {render, screen} from '@testing-library/vue';
import DropdownSettingsHeader from "./DropdownSettingsHeader.vue";

const title = "title"

function renderHeader() {
  const options = {
    props: {
      title,
    },
  }
  return render(DropdownSettingsHeader, options)
}

it('renders with title', () => {
  renderHeader()

  screen.getByText(title)
})

it('renders correctly', () => {
  const {html} = renderHeader()

  expect(html()).toMatchSnapshot()
})
