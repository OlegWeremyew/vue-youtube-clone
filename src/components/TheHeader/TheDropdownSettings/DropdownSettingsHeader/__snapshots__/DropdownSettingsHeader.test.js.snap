// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<section class=\\"flex border-b border-gray-200 text-black\\"><button type=\\"button\\" class=\\"px-3 focus:outline-none\\"><svg class=\\"w-6 h-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 20 20\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-testid=\\"base-icon\\">
      <path fill-rule=\\"evenodd\\" d=\\"M9.707 16.707a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414l6-6a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l4.293 4.293a1 1 0 010 1.414z\\" clip-rule=\\"evenodd\\"></path>
    </svg></button><span class=\\"py-3\\">title</span></section>"
`;
